sidewalk-labs.md

```diff
@@ -0,0 +1,95 @@
+### Who am I?
+
+I'm [Brett Naul](https://github.com/bnaul).
+I work at [Sidewalk Labs](https://www.sidewalklabs.com/).
+
+
+### What problem am I trying to solve?
+
+My team @ Sidewalk ("Model Lab") uses machine learning models to study human
+travel behavior in cities and produce high-fidelity simulations of the travel
+patterns/volumes in a metro area. Our process has three main steps:
+
+-   Construct a "synthetic population" from census data and other sources of
+    demographic information; this population is statistically representative of
+    the true population but contains no actual identifiable individuals.
+
+-   Train machine learning models on anonymous mobile location data to
+    understand behavioral patterns in the region (what times do people go to
+    lunch, what factors affect an individual's likelihood to use public
+    transit, etc.).
+
+-   For each person in the synthetic population, generate predictions from
+    these models and combine the resulting into activities into a single
+    model of all the activity in a region.
+
+
+### How Dask Helps
+
+Generating activities for millions of synthetic individuals is extremely
+computationally intensive; even with for example, a 96 core instance,
+simulating a single day in a large region initially took days. It was important
+to us to be able to run a new simulation from scratch overnight, and scaling to
+hundreds/thousands of cores across many workers with Dask let us accomplish our
+goal.
```

Comment:
I'm curious what tools within Dask are you using?  Is this all custom stuff with dask.delayed / futures?  All embarrassingly parallel?

Reply:
I think our uses fall into two categories:
1) Preprocessing: read in a whole bunch of large CSVs w/ dask.dataframe; repartition and group by some key; do some filtering and processing; save the output
2) "Machine learning": read in a large CSV in partitions; group by some key; apply SomeModel.train() or .predict() on each group; save the output

I've found groupby/apply to be a little more finicky so I tend to use a more manual approach, either using .map_partitions() or something like chunks = list(client.scatter(ddf.dask).values()); client.map(f, chunks)

Reply:
Cool.  That's valuable to mention.  To keep things high-to-medium-level I'm inclined to mention the bullet points but avoid the actual syntax used.  

However I recognize that this also skips over what I'm learning are nuanced flaws in dask.dataframe.  It would be good to get to the bottom of what was going on with groupby-apply at some point.